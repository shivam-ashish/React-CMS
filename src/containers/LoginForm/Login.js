import React, { Component } from 'react';
import MDSpinner from 'react-md-spinner';
import classes from './LoginForm.module.scss';
import BtnClass from '../../commonComponents/Button/Button.module.scss';
import Fire from '../../config/fire';
import Button from '../../commonComponents/Button/Button';

class Login extends Component {
  constructor(props) {
    super(props);
    this.login = this.login.bind(this);
    this.handleChange = this.handleChange.bind(this);
    this.state = {
      email: '',
      password: '',
      spinner: false,
    };
  }

  login = (e) => {
    const { email, password } = this.state;
    this.setState({ spinner: true });
    e.preventDefault();
    Fire.auth().signInWithEmailAndPassword(email, password)
      .then(() => {
      })
      .catch((error) => {
        alert(error.message);
        this.setState({ spinner: false });
      });
  }

  handleChange = (e) => {
    this.setState({ [e.target.name]: e.target.value });
  }

  render() {
    const { email, password, spinner } = this.state;
    const { toggle } = this.props;
    return (
      <div className={classes.loginForm}>
        <h1 className={classes.heading}>Login Here</h1>
        <label className={classes.labelField} htmlFor="email">Email Address : </label>
        <input className={classes.inputField} value={email} onChange={this.handleChange} type="email" name="email" />
        <label className={classes.labelField} htmlFor="email">Password : </label>
        <input className={classes.inputField} value={password} onChange={this.handleChange} type="password" name="password" />
        <Button
          type="Login"
          className={BtnClass.login}
          click={(evt) => {
            this.login(evt);
          }}
        >
          {spinner ? <MDSpinner /> : 'Login'}
        </Button>
        <Button
          type="New SignUp"
          className={BtnClass.signup}
          click={toggle}
        >
          {'+'}
        </Button>
      </div>
    );
  }
}

export default Login;
