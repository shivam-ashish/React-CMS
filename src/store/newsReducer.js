const initialState = {
  news: null,
};

const newsReducer = (state = initialState, { payload }) => {
  return {
    ...state,
    news: payload,
  };
};

export default newsReducer;
