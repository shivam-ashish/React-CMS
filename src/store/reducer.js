
const initialState ={
    isLoggedIn: false,
    user: 'user1',
}

const reducer = ( state = initialState, action ) => {
    const newState = { ...state };
    switch (action.type){
    case "LOGIN_STATE":
    return{
        ...state,
        isLoggedIn: Boolean,
        history: state.history.concat({
            isLoggedIn: Boolean
        })
    };
    case "UPDATE_USER":
    return{
        ...state,
        
        history: state.history.concat({
            
        })
    };
    default:
    return newState;
}
};

export default reducer;