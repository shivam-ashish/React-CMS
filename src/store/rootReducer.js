import { createStore, combineReducers } from 'redux';
import reducer from './reducers/reducer';
import newsReducer from './reducers/newsReducer';
import blogReducer from './reducers/blogReducer';

const rootReducer = combineReducers({
  reducer,
  newsReducer,
  blogReducer,
});

const store = createStore(rootReducer);

export default store;
