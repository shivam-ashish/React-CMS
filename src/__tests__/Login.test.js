import React from 'react';
import { mount } from 'enzyme';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import Login from '../containers/LoginForm/Login';
import store from '../store/rootReducer';

let wrapped;

beforeEach(() => {
  wrapped = mount(<Provider store={store}><Router><Login props="xyz" /></Router></Provider>);
});
afterEach(() => {
  wrapped.unmount();
});

it('has a Login component structure', () => {
  expect(wrapped.find('div').length).toEqual(1);
  expect(wrapped.find('h1').length).toEqual(1);
  expect(wrapped.find('input').length).toEqual(2);
  expect(wrapped.find('label').length).toEqual(2);
  expect(wrapped.find('Button').length).toEqual(2);
});

it('has text fields where users can type in', () => {
  wrapped.find('#email').simulate('change', {
    target: { value: 'username', name: 'email' },
  });
  wrapped.find('#password').simulate('change', {
    target: { value: 'password', name: 'password' },
  });
  wrapped.update();
  expect(wrapped.find('#email').prop('value')).toEqual('username');
  expect(wrapped.find('#password').prop('value')).toEqual('password');
});
