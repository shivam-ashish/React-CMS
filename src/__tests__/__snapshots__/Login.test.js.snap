// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it matches the snapshot 1`] = `
<div
  className="loginForm"
>
  <h1
    className="heading"
  >
    Login Here
  </h1>
  <label
    className="labelField"
    htmlFor="email"
  >
    Email Address : 
  </label>
  <input
    className="inputField"
    id="email"
    name="email"
    onChange={[Function]}
    type="email"
    value=""
  />
  <label
    className="labelField"
    htmlFor="email"
  >
    Password : 
  </label>
  <input
    className="inputField"
    id="password"
    name="password"
    onChange={[Function]}
    type="password"
    value=""
  />
  <button
    className="login"
    click={[Function]}
    onClick={[Function]}
    type="Login"
  >
    Login
  </button>
  <button
    className="signup"
    type="New SignUp"
  >
    +
  </button>
</div>
`;
